impeller/renderer/backend/vulkan/formats_vk.h

```diff
@@ -466,7 +467,9 @@ constexpr vk::AttachmentDescription CreateAttachmentDescription(
   switch (kind) {
     case AttachmentKind::kColor:
       vk_attachment.initialLayout = current_layout;
-      vk_attachment.finalLayout = vk::ImageLayout::eColorAttachmentOptimal;
+      vk_attachment.finalLayout =
+          resolve_texture ? vk::ImageLayout::ePresentSrcKHR
+                          : vk::ImageLayout::eColorAttachmentOptimal;
```

Comment:
Is this the right switch value?

Spec says that eColorAttachmentOptimal "must only" be used as a color ro resolve in a `VkFramebuffer`. 

Perhaps the argument should instead be `bool is_framebuffer` or something? Or perhaps an enum where the enumerated values are something like `kFramebuffer` and `kNonFramebuffer`.

Reply:
I'm not sure I really understand what `ePresentSrcKHR` is really for. Did you already try `eGeneral`?

Reply:
I don't really understand either, but it was the value used in https://vulkan-tutorial.com/Multisampling so I just went with it. I'll try general

Reply:
eGeneral works! lets see if it validates